SharedSource/UnitTests/FunctionalProgramming/EnumerableExtensionsTest.cs

```diff
@@ -203,10 +203,10 @@ public void CreateSequence_WhilePredicateEvaluatesTrue_WithSingleElement ()
     }
 
     [Test]
-    public void CreateSequence_WhilePredicateEvaluatesTrue_WithValueType ()
+    public void CreateSequence_WhilePredicateEvaluatesTrue_StopsOnNull ()
     {
-      IEnumerable<int> actual = 4.CreateSequence (e => e - 1, e => e > 0);
-      Assert.That (actual.ToArray(), Is.EqualTo (new[] { 4, 3, 2, 1 }));
+      IEnumerable<string> actual = "start".CreateSequence (e => null, e => e != "end");

```

Comment:
hmm.. i think we can make an a bit more involved test.

Start-string: "ABCD"
Next Element:  value.SubString(0, len-1) and if string.length == 0 => null. 
The PRedicate can be "true"
Assert is then "ABCD", ABC, AB, A